gematria/datasets/python/exegesis_benchmark.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string_view>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "gematria/datasets/exegesis_benchmark_lib.h"
+#include "gematria/datasets/find_accessed_addrs.h"
+#include "gematria/llvm/llvm_to_absl.h"
+#include "llvm-c/Target.h"
+#include "llvm/tools/llvm-exegesis/lib/PerfHelper.h"
+#include "llvm/tools/llvm-exegesis/lib/TargetSelect.h"
+#include "pybind11/detail/common.h"
+#include "pybind11/pybind11.h"
+#include "pybind11_abseil/import_status_module.h"
+#include "pybind11_abseil/status_casters.h"  // IWYU pragma: keep
+
+namespace gematria {
+
+namespace py = ::pybind11;
+
+using namespace llvm;
+using namespace llvm::exegesis;
+
+PYBIND11_MODULE(exegesis_benchmark, m) {
+  m.doc() = "Code for benchmarking basic blocks.";
+
+  py::google::ImportStatusModule();
+
+  py::class_<ExegesisBenchmark>(m, "ExegesisBenchmark")
+      .def("create",
+           []() -> absl::StatusOr<std::unique_ptr<ExegesisBenchmark>> {
+             // LLVM Setup
+             LLVMInitializeX86TargetInfo();
+             LLVMInitializeX86TargetMC();
+             LLVMInitializeX86Target();
+             LLVMInitializeX86AsmPrinter();
+             LLVMInitializeX86AsmParser();
+             LLVMInitializeX86Disassembler();
+
+             // Exegesis Setup
+             InitializeX86ExegesisTarget();
+
+             if (pfm::pfmInitialize())
+               return absl::InternalError("Failed to initialize PFM");
+
+             return LlvmExpectedToStatusOr(ExegesisBenchmark::create());
+           })
+      .def("process_annotated_block",
+           [](ExegesisBenchmark& Self, std::string_view BlockHex,
+              const BlockAnnotations& Annotations) {
+             return LlvmExpectedToStatusOr(
+                 Self.processAnnotatedBlock(BlockHex, Annotations));
```

Comment:
Is there a binding for `llvm::exegesis::BenchmarkCode`? If not, do we need one?

Reply:
Added a binding. For now, I didn't add any methods as the python side shouldn't need to access anything inside.